lib/std/mem.zig

```diff
@@ -3835,6 +3835,7 @@ fn CopyPtrAttrs(
     comptime source: type,
     comptime size: std.builtin.Type.Pointer.Size,
     comptime child: type,
+    comptime sentinel: ?*const anyopaque,
```

Comment:
Seems to me like this should be `comptime sentinel: ?child`

Reply:
I tried that first, but ran into a bunch of weird errors like `error: comptime dereference requires '[:0]u8' to have a well-defined layout` (when a sentinel is present), and `error: sentinels are only allowed on slices and unknown-length pointers` (when no sentinel).

I think there's something about how sentinels are represented at comptime that interferes with that, but I don't understand it.  Can you make that work?

Reply:
I see that this same code is present in the previous https://github.com/ziglang/zig/pull/18021 .
The `comptime deference requires [...] well-defined layout` error (or at least positives) should have been resolved by https://github.com/ziglang/zig/pull/19630 , which was only merged on 04-17.
If you haven't since then, please re-check whether you still run into the issue or whether it's since been fixed.

Reply:
Yes.  Sorry I should have been clearer - I tried it originally with the first PR, and I tried it again after @InKryption commented and those are the errors I currently get (when trying `comptime sentinel: ?child`).

Reply:
Now I'm doubting myself - let me recheck.

Reply:
Yes I get those errors when trying `comptime sentinel: ?child` with zig version `0.13.0-dev.211+6a65561e3`

Reply:
The compile errors here seem incorrect. Would you be able to make a commit with the proposed change which triggers the compile error, so I can take a look and file a compiler bug if one exists?

Reply:
I've committed that change, and it now compiles without error!  So this tangent is resolved.  Thanks for pushing me to retry it!